astropy/io/fits/column.py

```diff
@@ -2402,3 +2450,66 @@ def _convert_ascii_format(format, reverse=False):
             recformat += str(width)
 
         return recformat
+
+
+def _parse_tdisp_format(tdisp):
+    """
+    Parse the ``TDISPn`` keywords for ASCII and binary tables into a
+    ``(format, width, precision, exponential)`` tuple (the TDISP values
+    for ASCII and binary are identical except for 'Lw', which is only present in BINTABLE
+    extensions
+    """
+
+    # Use appropriate regex for format type
+    tdisp = tdisp.strip()
+    fmt_key = tdisp[0] if tdisp[0] !='E' or tdisp[1] not in 'NS' else tdisp[:2]
+    try:
+        tdisp_re = TDISP_RE_DICT[fmt_key]
+    except KeyError:
+        raise VerifyError('Format {!r} is not recognized.'.format(tdisp))
+
+
+    match = tdisp_re.match(tdisp.strip())
+    if not match or match.group('formatc') is None:
+        raise VerifyError('Format {!r} is not recognized.'.format(tdisp))
+
+    formatc = match.group('formatc')
+    width = match.group('width')
+    precision = None
+    exponential = None
+
+    # Some formats have
+    if tdisp[0] in ('I', 'B', 'O', 'Z', 'F', 'E', 'G', 'D'):
+        precision = match.group('precision')
+        if precision is None:
+            precision = 1
+    if tdisp[0] in ('E', 'D', 'G') and tdisp[1] not in ('N', 'S'):
+        exponential = match.group('exponential')
+        if exponential is None:
+            exponential = 1
+
+
+    # Once parsed, check format dict to do conversion to a formatting string
+
+
+    return formatc, width, precision, exponential
+
+
+def _fortran_to_python_format(tdisp):
+    """
+    Turn the TDISPn fortran format pieces into a final Python format string.
+    See the format_type definitions below. If codes is changed to take
+    advantage of the exponential specification, will need to add it as another
+    input parameter.
+
+    """
+    format_type, width, precision, exponential = _parse_tdisp_format(tdisp)
+
+    try:
+        fmt = TDISP_FMT_DICT[format_type]
+        if format_type not in ['EN', 'ES']:
+            return fmt.format(width=width, precision=precision)
+        else:
+            return fmt
```

Comment:
Ehhh, I'm gonna reverse this if else to check for in ['EN', 'ES'], not sure why I did it that way around...